agent/main.go

```diff
@@ -159,8 +159,10 @@ func main() {
 			},
 		},
 		SkipInstallation: skipInstallation,
-		DownloadPath:     downloadpath,
 		Recorder:         mgr.GetEventRecorderFor("hostagent-controller"),
+		Installer: &reconciler.K8sInstaller{
```

Comment:
You should assign a reference of `installer.New` to Installer.

Instead of doing a `installer.New` in the `Install` & `Uninstall` functions you've defined below, it should be instantiated in `main.go` itself. Otherwise, we still have the same concern of instantiating installer multiple times and not actually taking the benefit of pre-checks.